staging/src/k8s.io/legacy-cloud-providers/aws/aws_loadbalancer.go

```diff
@@ -1539,3 +1541,29 @@ func (c *Cloud) findInstancesForELB(nodes []*v1.Node) (map[InstanceID]*ec2.Insta
 
 	return instances, nil
 }
+
+// filterTargetNodes uses node labels to filter the nodes that should be targeted by the ELB,
+// checking if at least one of the target node labels provided in an annotation, is present in the node
+func filterTargetNodes(nodes []*v1.Node, annotations map[string]string) []*v1.Node {
+
+	targetNodeLabels := getKeyValuePropertiesFromAnnotation(annotations, ServiceAnnotationLoadBalancerTargetNodeLabels)
+
+	if len(targetNodeLabels) == 0 {
+		return nodes
+	}
+
+	targetNodes := make([]*v1.Node, 0)
```

Comment:
better to be `var targetNodes []v1.Node` for an empty slice, or `targetNodes := make([]*v1.Node, 0, len(nodes))` for a slice predicated to have same length of nodes possibly.